### PR TITLE
Add SparkDataType as wrapper for unmapped spark data type

### PR DESCRIPTION
**Pull request by penghuo**
### Description
Add SparkDataType as wrapper for unmapped spark data type
 
### Issues Resolved
* https://github.com/opensearch-project/sql/issues/2367
* https://github.com/opensearch-project/sql/issues/2408
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: spark/src/main/java/org/opensearch/sql/spark/functions/response/DefaultSparkSqlFunctionResponseHandle.java [hunk 1]
```diff
@@ -64,30 +67,43 @@ private static LinkedHashMap<String, ExprValue> extractRow(
...
+          // TODO :: correct this to ExprTimestampValue
+          linkedHashMap.put(column.getName(), new ExprStringValue(row.getString(column.getName())));
+        } else if (type == ExprCoreType.TIMESTAMP) {
+          linkedHashMap.put(
+              column.getName(), new ExprTimestampValue(row.getString(column.getName())));
+        } else if (type == ExprCoreType.STRING) {
+          linkedHashMap.put(column.getName(), new ExprStringValue(row.getString(column.getName())));
+        } else {
+          // SparkDataType
+          Object jsonValue = row.get(column.getName());
```
Comment (dai-chen) [inline spark/src/main/java/org/opensearch/sql/spark/functions/response/DefaultSparkSqlFunctionResponseHandle.java:98]: I forgot earlier how we added support for non-json response such as Spark EXPLAIN output. This doesn't impact that right?

**Review by dai-chen — APPROVED** [review]
Thanks for the fix!

File: spark/src/main/java/org/opensearch/sql/spark/functions/response/DefaultSparkSqlFunctionResponseHandle.java [hunk 2]
```diff
@@ -64,30 +67,43 @@ private static LinkedHashMap<String, ExprValue> extractRow(
...
+          linkedHashMap.put(column.getName(), new ExprFloatValue(row.getFloat(column.getName())));
+        } else if (type == ExprCoreType.DATE) {
+          // TODO :: correct this to ExprTimestampValue
+          linkedHashMap.put(column.getName(), new ExprStringValue(row.getString(column.getName())));
+        } else if (type == ExprCoreType.TIMESTAMP) {
+          linkedHashMap.put(
+              column.getName(), new ExprTimestampValue(row.getString(column.getName())));
+        } else if (type == ExprCoreType.STRING) {
+          linkedHashMap.put(column.getName(), new ExprStringValue(row.getString(column.getName())));
+        } else {
```
Comment (vamsi-amazon) [inline spark/src/main/java/org/opensearch/sql/spark/functions/response/DefaultSparkSqlFunctionResponseHandle.java:96]: Just for understanding: anything apart from above datatypes is a SparkDataType.

**Comment by vamsi-amazon** [comment]
Thanks for the fix. LGTM

**Review by vamsi-amazon — APPROVED** [review]

File: spark/src/main/java/org/opensearch/sql/spark/functions/response/DefaultSparkSqlFunctionResponseHandle.java (hunk 2 above)
Comment (penghuo) [reply spark/src/main/java/org/opensearch/sql/spark/functions/response/DefaultSparkSqlFunctionResponseHandle.java:96]: yes, any spark datatype sql can not mapping to is wrapped as SparkDataType.

The major reason for doing this mapping is event analysis frontend feature use SQL data type to display type icon.

File: spark/src/main/java/org/opensearch/sql/spark/functions/response/DefaultSparkSqlFunctionResponseHandle.java (hunk 1 above)
Comment (penghuo) [reply spark/src/main/java/org/opensearch/sql/spark/functions/response/DefaultSparkSqlFunctionResponseHandle.java:98]: explain result is string value, 

add IT https://github.com/opensearch-project/sql/pull/2492/files#diff-c53435443207a93d99630ec47fde52a91da4e4403b712c9e3bb51aa074b51ce6R335

**Review by dai-chen — APPROVED** [review]

**Review by vamsi-amazon — APPROVED** [review]

**Pull request merged**